cataclysmbnteam/Cataclysm-BN
# feat: tags generation with lookup support

#### Summary

SUMMARY: Infrastructure "Tool to generate ctags for faster JSON navigation"

#### Purpose of change

https://github.com/cataclysmbnteam/Cataclysm-BN/assets/54838975/c2167acb-2b1b-4b7c-817a-a4878c32eda0

provide an easy way to nagivate JSON ids using ctags

#### Describe the solution

ported some of #2623 

#### Testing

1. run `make ctags` to generate `.tags` file under `.vscode/` directory.
2. rename its name to `.tags.bak` for future reference.
3. refactor related typescript code.
4. run `deno test --allow-read` to check if new changes generate different tags.

#### Additional context

[deno](https://deno.land/) is required to run the scripts.
